extensions/wikia/Sitemap/SpecialSitemap_body.php

```diff
@@ -69,7 +69,15 @@ public function __construct( $name = 'Sitemap' ) {
 	 * @param $subpage Mixed: subpage of SpecialPage
 	 */
 	public function execute( $subpage ) {
-		global $wgMemc, $wgRequest, $wgOut;
+		global $wgMemc, $wgRequest, $wgOut, $wgEnableSpecialSitemapExt;
+
+		if ( strpos( $subpage, '-newsitemapxml-' ) !== false ) {
+			if ( empty( $wgEnableSpecialSitemapExt ) ) {
+				$this->print404();
+				return;
+			}
+			return ( new SpecialSitemapXmlController() )->execute( $subpage );
+		}
 
 		if ( !is_array( $wgMemc->get( wfMemcKey( 'sitemap-index' ) ) ) ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "sitemap-index" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)